### PR TITLE
Add schedule fields and default engine to the formik database form

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/26178

How to test:
- `yarn storybook` -> `DatabaseForm`
- H2 -> Advanced options -> Choose when syncs and scans happen
- Make sure it is possible to change schedules and they are reflected in submit values

Screenshot 2022-12-01 at 14 56 29

### CODE REVIEW

File: frontend/src/metabase/databases/components/DatabaseEngineField/DatabaseEngineField.tsx [hunk 1]
```diff
@@ -28,7 +30,7 @@ const DatabaseEngineField = ({
   return isAdvanced ? (
     <DatabaseEngineSelect
       options={options}
-      disabled={!isNew}
+      disabled={values.is_sample}
```
Comment (ranquild) [inline frontend/src/metabase/databases/components/DatabaseEngineField/DatabaseEngineField.tsx:33]: Not related to this PR but decided to fix it here

File: frontend/src/metabase/components/SchedulePicker/SchedulePicker.tsx [hunk 2]
```diff
@@ -66,7 +66,7 @@ type ScheduleChangeProp = { name: ScheduleProperty; value: unknown };
 export interface SchedulePickerProps {
   schedule: ScheduleSettings;
   scheduleOptions: ScheduleType[];
-  timezone: string;
+  timezone?: string;
```
Comment (ranquild) [inline frontend/src/metabase/components/SchedulePicker/SchedulePicker.tsx:69]: Not used unless `textBeforeSendTime` is used

File: frontend/src/metabase/databases/types.ts [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
   name: string;
   engine: string | undefined;
   details: Record<string, unknown>;
   schedules: DatabaseSchedules;
   auto_run_queries: boolean;
   refingerprint: boolean;
+  is_sample: boolean;
+  is_full_sync: boolean;
+  is_on_demand: boolean;
 }
```
Comment (kulyk) [inline frontend/src/metabase/databases/types.ts:20]: Should we consider extending the [`Database` type from `metabase-types/api`](https://github.com/metabase/metabase/blob/b6e50930239b9cb17c69ee289cba486581d7a442/frontend/src/metabase-types/api/database.ts#L12) and building `DatabaseValues` by stuff like `Pick<Database, ...>` ?

**Review by kulyk — APPROVED** [review]
🕒 G T M

File: frontend/src/metabase/databases/types.ts (hunk 3 above)
Comment (ranquild) [reply frontend/src/metabase/databases/types.ts:20]: Agree, I'll reconsider this in one of the future PRs. There is some duplication between `Database`, `DatabaseInfo` and `DatabaseValues` we need to solve

**Pull request merged**